pkg/image/containerdregistry/resolver.go

```diff
@@ -55,9 +50,6 @@ func NewResolver(configDir string, insecure bool, roots *x509.CertPool) (remotes
 		)),
 		docker.WithClient(client),
 	}
-	if insecure {
```

Comment:
Question: Have you checked your changes still work with http endpoints?
I'm not sure what these `regopts` do exactly. Should we maybe still keep this but only add if if the url scheme is http?
@njhale any thoughts here?

Reply:
I have not checked that, good point. Didn't think about that.

But if I do think of it, why would someone pass `--skip-tls` when using HTTP in the first place? And also, how does one even specify http registries? e.g. How would you tell docker to use http to communicate with `quay.io/foo/bar:latest`? It uses https by default. Searching online didn't lead me to a conclusive answer

Reply:
Also, worth mentioning that seems like `WithPlainHTTP` configures the `plainHTTP` option which is deprecated (although the function WithPlainHTTP itself isn't deprecated which is odd)

Reply:
Here's how it's done:
https://docs.docker.com/registry/insecure/#deploy-a-plain-http-registry

Basically a config file lists all registries where docker should try to communicate with HTTPS, if it gets a cert, ignore any errors, if there's no HTTPS on that server fallback to HTTP.

Doesn't seem like you can specify the protocol in the image address itself in any way I could find

Reply:
I agree with you here. --skip-tls and http are not related at all. I just don't know if these regopts we are removing here are needed for something specific (e.g. pipeline). We may still need to add a --use-http options or something like that. So, I'll set it to ok-to-test but I'd like @njhale (who I think originally added this) to ok.

Reply:
Original author is here https://github.com/operator-framework/operator-registry/commit/a16399f05c380ba432691d2e1d99a69aa5a269e2

Reply:
We can't change/remove current `--skip-tls` functionality, but we can deprecate the flag and add new flags `--skip-tls-verify` and `--use-http` to enable their respective namesake's functionality.

Reply:
Assuming @omertuc does not update this PR soon, @dinhxuanvu @jpower432 can one of you take on the changes in a separate PR?

Reply:
Ah didn't realize your bullets were concrete change suggestions.

Yeah please take it on yourselves as I don't feel confident enough in this code base to make those changes

Reply:
Sounds good, thanks for all you've done so far!

Reply:
I can make these changes.